drivers/rtc/rtc_emul.c

```diff
@@ -277,42 +265,41 @@ static int rtc_emul_set_time(const struct device *dev, const struct rtc_time *ti
 		return -EINVAL;
 	}
 
-	k_mutex_lock(&data->lock, K_FOREVER);
-
-	data->datetime = (*timeptr);
-	data->datetime.tm_isdst = -1;
-	data->datetime.tm_nsec = 0;
+	K_SPINLOCK_GUARDED_SECTION(&data->lock)
+	{
+		data->datetime = (*timeptr);
+		data->datetime.tm_isdst = -1;
+		data->datetime.tm_nsec = 0;
 
-	data->datetime_set = true;
-
-	k_mutex_unlock(&data->lock);
+		data->datetime_set = true;
+	}
 
 	return 0;
 }
 
 static int rtc_emul_get_time(const struct device *dev, struct rtc_time *timeptr)
 {
 	struct rtc_emul_data *data = (struct rtc_emul_data *)dev->data;
+	int ret = 0;
 
 	/* Validate arguments */
 	if (timeptr == NULL) {
 		return -EINVAL;
 	}
 
-	k_mutex_lock(&data->lock, K_FOREVER);
+	K_SPINLOCK_GUARDED_SECTION(&data->lock)
+	{
+		/* Validate RTC time is set */
+		if (data->datetime_set == false) {
+			ret = -ENODATA;
 
-	/* Validate RTC time is set */
-	if (data->datetime_set == false) {
-		k_mutex_unlock(&data->lock);
+			K_SPINLOCK_LEAVE_GUARDED_SECTION;
+		}
 
-		return -ENODATA;
+		(*timeptr) = data->datetime;
```

Comment:
Technically no need for parenthesis, so it can just be `*timeptr`

Reply:
Fixed in a separate commit.